contents/tutorials/astro-analytics.md

```diff
@@ -0,0 +1,386 @@
+
+![Custom event capture](../images/tutorials/astro-analytics/praise.png)
+
+## Setting up a feature flag
+
+To use a feature flag in your app, first, you must create it in PostHog. Go to the [feature flags tab](https://app.posthog.com/feature_flags), click "New feature flag," add a key (we choose `new-button`), set the release condition to 100% of users, and click "Save."
+
+With the feature flag, go back to your home page at `pages/index.astro`. Add to your script the `onFeatureFlags()` function with an `isFeatureEnabled()` for your `new-button` flag. If enabled, change the `innerHTML` of your button.
+
+```js
+---
+// src/pages/index.astro
+import Layout from '../layouts/Layout.astro';
+---
+<Layout>
+	<h1>Home</h1>
+	<button class="main">Great site!</button>
+	<script>
+		const button = document.querySelector('.main');
+		button.addEventListener('click', () => {
+			window.posthog.capture('praise_received')
+		});
+
+		window.posthog.onFeatureFlags(() => {
+			if (window.posthog.isFeatureEnabled('new-button')) {
+				button.innerText = 'The best site ever!';
+    	}
+		});
+	</script>
+</Layout>
+```
+
+When you reload your page, it shows different button text controlled by the PostHog feature flag.
+
+![Flag](../images/tutorials/astro-analytics/flag.png)
+
+## Adding PostHog on the server side
+
+So far, we have only used PostHog on the client side. We can also set up PostHog on the server side to capture events on page load, evaluate feature flags before page loads, and more. To do this, start by installing the [Node SDK](/docs/libraries/node) by running:
+
+```bash
+npm i posthog-node
+```
+
+In the `src` folder, create a `posthog.js` file. This is where we set up the code to create the PostHog Node client using our project API key and instance address.
+
+```js
+// src/posthog.js
+import { PostHog } from 'posthog-node';
+
+let posthogClient = null;
+
+export default function PostHogClient() {
+  if (!posthogClient) {
+    posthogClient = new PostHog('<ph_project_api_key>', {
+      host: '<ph_instance_address>',
+    });
+  }
+  return posthogClient;
+}
+```
+
+To show how we can use this, we import it into our home `pages/index.astro` file and then capture an event with it.
+
+```js
+---
+// src/pages/index.astro
+import Layout from '../layouts/Layout.astro';
+import PostHogClient from '../posthog.js';
+
+const phClient = PostHogClient();
+phClient.capture(
+	{
+		event: 'server_side_event', 
+		distinctId: 'ian@posthog.com'
+	}
+);
+---
+<!-- The rest of the component -->
+```
+
+Now when we reload the page, PostHog captures a `server_side_event`.
+
+## Identifying users to connect server and client
+
+One issue with our implementation is the hard-coded user distinct ID we are using to capture events. For it to use a real user’s ID, we can either:
+
+1. Use the distinct ID set by the client side PostHog in the request cookie.
+2. Create a new distinct ID on the server and use `identify()` to connect it in the client.
```

Comment:
Worth checking if the hyperlink renders correctly. I cant remember whats the correct order for using them in codeblocks
```suggestion
2. Create a new distinct ID on the server and use [`identify()`](/docs/product-analytics/identify) to connect it in the client.
```